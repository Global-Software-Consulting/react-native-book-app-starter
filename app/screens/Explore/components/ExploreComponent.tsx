import React, {useState, useEffect} from 'react';
import {View,TouchableHighlight,FlatList, Alert, BackHandler} from 'react-native';
import {Text} from 'react-native-paper';
import { useRoute } from '@react-navigation/native';

import NavigationService from './../../../navigation/NavigationService';
//importing card component
import BookCard from './../../../components/BookCard/BookCard';
import {useStyles} from '../styles';

interface Props {
  books?:[];
  name?:string;
base_url?:string;
}

const ExploreComponent: React.FC<Props> = ({books, name, base_url}) => 
    {
//theme handling
const styles = useStyles();

  return (
<View>
      <Text style={styles.nameStyle}>Hi {name} </Text>
      <Text style={styles.tagLineStyle}>Let's find something new </Text>

      <View style={styles.horizontalRuler} /> 

      <Text style={styles.listCaptionStyle}>Trending</Text>
      <FlatList horizontal
       data={books?.filter((item) => { return item.averageRating > 3 })}
       contentContainerStyle={styles.flatList}
       renderItem={({ item, index }) => (
       <TouchableHighlight
        key={item}
        underlayColor='grey'
        onPress={()=>{ NavigationService.navigate('BookDetail',item)}}  >
          <BookCard url={'https://edit.org/images/cat/book-covers-big-2019101610.jpg'} isFavorite={true} styleSelect='Custom' bookTitle={item.title} />
      </TouchableHighlight>
      )}
  showsHorizontalScrollIndicator={false}    
/>

      <View style={styles.horizontalRuler} /> 

      <Text style={styles.listCaptionStyle}>New Releases</Text>
      <FlatList horizontal
       data={books?.filter((item) => { return item.averageRating < 3 })}
       contentContainerStyle={styles.flatList}
       renderItem={({ item, index }) => (
       <TouchableHighlight
        key={item}
        underlayColor='grey'
        onPress={()=>{ NavigationService.navigate('BookDetail',item)}}  >
          <BookCard url={'https://edit.org/images/cat/book-covers-big-2019101610.jpg'} isFavorite={false} styleSelect='General' bookTitle={item.title}  />
      </TouchableHighlight>
      )}
      showsHorizontalScrollIndicator={false}    
    />

      <View style={styles.horizontalRuler} /> 

      <Text style={styles.listCaptionStyle}>Selected for you</Text>
      <FlatList horizontal
       data={books?.filter((item) => {return item.averageRating == 0})}
       contentContainerStyle={styles.flatListLast}
       renderItem={({ item, index }) => (
       <TouchableHighlight
        key={item}
        underlayColor='grey'
        onPress={()=>{ NavigationService.navigate('BookDetail',{item})}}  >
          <BookCard url={'https://edit.org/images/cat/book-covers-big-2019101610.jpg'} isFavorite={false} styleSelect='General'  bookTitle={item.title} />
      </TouchableHighlight>
      )}
      showsHorizontalScrollIndicator={false}    
    />
    </View>

  )
    };

export default ExploreComponent;
