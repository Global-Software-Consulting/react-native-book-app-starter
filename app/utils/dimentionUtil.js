import { useDeviceOrientation } from '@react-native-community/hooks';
import { useCallback, useEffect, useMemo, useState } from 'react';
import { Dimensions, PixelRatio } from 'react-native';

const widthPercentageToDP = (screenWidth, widthPercent) => {
    const elemWidth = typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent);
    return PixelRatio.roundToNearestPixel((screenWidth * elemWidth) / 100);
};

const heightPercentageToDP = (screenHeight, heightPercent) => {
    const elemHeight =
        typeof heightPercent === 'number' ? heightPercent : parseFloat(heightPercent);
    return PixelRatio.roundToNearestPixel((screenHeight * elemHeight) / 100);
};

export function getPercentageWidth() {
    const [width, setWidth] = useState(Dimensions.get('window').width);
    useEffect(() => {
        Dimensions.addEventListener('change', (newDimensions) => {
            // Retrieve and save new dimensions
            screenWidth = newDimensions.window.width;
            setWidth(screenWidth);
        });
        return Dimensions.removeEventListener('change', () => {});
    }, []);
    const wp = useCallback((value) => widthPercentageToDP(width, value), [width]);

    return wp;
}
export function getPercentageHeight() {
    const [height, setHeight] = useState(Dimensions.get('window').height);
    useEffect(() => {
        Dimensions.addEventListener('change', (newDimensions) => {
            screenHeight = newDimensions.window.height;
            setHeight(screenHeight);
        });
        return Dimensions.removeEventListener('change', () => {});
    }, []);
    const hp = useCallback((value) => heightPercentageToDP(height, value), [height]);

    return hp;
}
