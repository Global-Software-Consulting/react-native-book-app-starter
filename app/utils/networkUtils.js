import NetInfo from "@react-native-community/netinfo";
export default class NetworkUtils {
  static async isNetworkAvailable() {
    const response = await NetInfo.fetch();
    return response.isConnected;
}}