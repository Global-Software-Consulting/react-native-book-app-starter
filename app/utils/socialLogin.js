import auth from '@react-native-firebase/auth';
import { LoginManager, AccessToken } from 'react-native-fbsdk';
import { GoogleSignin } from '@react-native-google-signin/google-signin';


export default class socialLogin {

static async loginWithGoogle() {
// Get the users ID token
const { idToken } = await GoogleSignin.signIn();

// Create a Google credential with the token
const googleCredential = auth.GoogleAuthProvider.credential(idToken);

// Sign-in the user with the credential
return auth().signInWithCredential(googleCredential);
}

static async loginWithFacebook() {
        // Attempt login with permissions
  const result = await LoginManager.logInWithPermissions(['public_profile', 'email']);

  if (result.isCancelled) {
    throw 'User cancelled the login process';
  }

  // Once signed in, get the users AccesToken
  const data = await AccessToken.getCurrentAccessToken();

  if (!data) {
    throw 'Something went wrong obtaining access token';
  }

  // Create a Firebase credential with the AccessToken
  const facebookCredential = auth.FacebookAuthProvider.credential(data.accessToken);

  // Sign-in the user with the credential
  return auth().signInWithCredential(facebookCredential);
}

}
