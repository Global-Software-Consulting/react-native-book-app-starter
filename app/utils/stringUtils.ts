export const getInitials = (name: string) => {
  let initials = Array.prototype.map
    .call(name.split(' '), function (x) {
      return x.substring(0, 1).toUpperCase();
    })
    .join('');
  return initials.substring(0, 2);
};
