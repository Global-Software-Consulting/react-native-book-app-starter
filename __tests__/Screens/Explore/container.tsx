import { useIsFocused } from '@react-navigation/native'
import { render } from '@testing-library/react-native'
import React from 'react'
import 'react-native-gesture-handler/jestSetup'
import * as redux from 'react-redux'
import { act } from 'react-test-renderer'
import ExploreComponent from 'screens/Explore/Container'
import { getPercentageHeight, getPercentageWidth } from 'utils/dimentionUtil'

const DeviceTypeUtilsMock = jest.requireMock('utils/dimentionUtil')

jest.mock('./../../../app/utils/dimentionUtil', () => {
    return {
        getPercentageHeight: jest.fn(),
        height: '100',
        getPercentageWidth: jest.fn(),
    }
})

describe('Screen test', () => {
    test('Snapshot with loading as true ', () => {
        const height = jest.fn()
        getPercentageHeight.mockReturnValue(jest.fn())
        const width = jest.fn()
        getPercentageWidth.mockReturnValue(jest.fn())
        act(() => {
            const state = {
                loadingReducer: { isLoading: false },
                loginReducer: { user: 'as' },
                appReducer: {
                    books: [
                        { averageRating: 2 },
                        { averageRating: 4 },
                        { averageRating: 5 },
                        { averageRating: 1 },
                        { averageRating: 3 },
                    ],
                },
            }
            jest.spyOn(redux, 'useSelector').mockImplementation((val) => {
                val(state)
                expect(tree).toMatchSnapshot()
            })
        })
        const isFocused = jest.fn()
        useIsFocused.mockReturnValue(true)

        const tree = render(<ExploreComponent />)
    })
})
