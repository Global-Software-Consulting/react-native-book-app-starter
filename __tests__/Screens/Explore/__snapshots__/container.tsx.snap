// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen test Snapshot with loading as true  1`] = `undefined`;

exports[`Screen test Snapshot with loading as true  2`] = `undefined`;
